packages/snaps-utils/src/json-rpc.ts

```diff
@@ -57,6 +57,43 @@ export function assertIsRpcOrigins(
   );
 }
 
+export const KeyringOriginsStruct = refine(
+  object({
+    allowedOrigins: optional(array(string())),
+  }),
+  'Keyring origins',
+  (value) => {
+    if (value.allowedOrigins && value.allowedOrigins.length > 0) {
```

Comment:
Is at least one origin required? @danroc

Reply:
Very good point, I think we could have the use case where the `allowedOrigins` is empty to indicate that no dapp can interact with the Account Snap.